sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/PayloadSizeGate.java

```diff
@@ -38,12 +38,18 @@
      * @return Buffered data or incoming data depending on threshold condition.
      */
     Flux<ByteBuffer> write(ByteBuffer buf) {
+        /* TODO (gapra): Investigate if there is way to avoid copying the data twice since this may result in lower
+            perf. The other copy is when we write the buffer to the pool */
+        /* Deep copy the buffer. This is required to ensure integrity of data. */
+        ByteBuffer cachedBuffer = ByteBuffer.allocate(buf.remaining()).put(buf);
+        cachedBuffer.flip();
```

Comment:
Ups. My bad.
I think there isn't double copy now as copying was removed buffer pool - hence regression.

For future. Maybe it would be good to extract copying part to separate component that's inserted into Flux transform early in the process - so that it's more explicit (as we apparently quite depend on that).

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10639